contracts/ERC7432/ERC7432.sol

```diff
@@ -80,17 +82,26 @@ contract ERC7432 is IERC7432 {
         uint256 _tokenId,
         address _revoker,
         address _grantee
-    ) external override onlyApproved(_tokenAddress, _tokenId, _revoker) {
-        _revokeRole(_role, _tokenAddress, _tokenId, _revoker, _grantee);
+    ) external override {
+        bool _isRevokerApproved = isRoleApprovedForAll(_tokenAddress, _revoker, msg.sender) || getApprovedRole(_tokenAddress, _tokenId, _revoker, msg.sender);
+        bool _isGranteeApproved = isRoleApprovedForAll(_tokenAddress, _grantee, msg.sender) || getApprovedRole(_tokenAddress, _tokenId, _grantee, msg.sender);
+        require(_isRevokerApproved || _isGranteeApproved, "ERC7432: sender must be approved");
+
+        address _caller = _isGranteeApproved ? _grantee : _revoker; // In case of operator being approved for both grantee and revoker, the grantee will be the caller.
+
+        _revokeRole(_role, _tokenAddress, _tokenId, _revoker, _grantee, _caller);
     }
 
     function _revokeRole(
         bytes32 _role,
         address _tokenAddress,
         uint256 _tokenId,
         address _revoker,
-        address _grantee
+        address _grantee,
+        address _caller
     ) internal {
+        bool _isRevocable = roleAssignments[_revoker][_grantee][_tokenAddress][_tokenId][_role].revocable;
+        require(_isRevocable || _caller == _grantee, "ERC7432: Account can't revoke this role");
```

Comment:
```suggestion
        require(_isRevocable, "ERC7432: Role is not revocable");
        require(_caller == _grantee, "ERC7432: Caller is not allowed to revoke this role");
```

Reply:
we can't separate a `OR` condition to two requires, because will become an `AND`